Getting error "[Makefile:13: migrate] Fehler 1"

Dear all,

I tried to install docker-fittrackee, but I got the following error:

![fittrackee_error](https://github.com/SamR1/docker-fittrackee/assets/92477674/d09031cf-b27f-413f-9ce2-eabc4dac4098)

It seems, that two dcoker containers start, but I can't connect to port 5000. This was the first installation from Fittrackee.

THX for this great software 

_Edit 20230528:_
I get the error during excecution excecute ` make build up init`

Hi,

so enclose my new .env. I also set `GUNICORN_LOG` back to the the default value again. The directory `/home/duncan/docker/docker-fittrackee/logs` is unfortunately still empty and the error is the same `make: *** [Makefile:13: migrate] Fehler 1`.

Any ideas here? I looks like it get stuck in the `migrate` part of the Makefile. Is this needed if I install it the first time?

```

# Custom variables initialisation
# All uncommented variables must be initialized

# Host directory to store FitTrackee logs (must be an absolute path)
FITTRACKEE_LOG_DIR=/home/duncan/docker/docker-fittrackee/logs

# Application
FLASK_APP=fittrackee
FLASK_SKIP_DOTENV=1
PORT=5000
APP_SETTINGS=fittrackee.config.ProductionConfig
APP_SECRET_KEY='my_secret_key'
APP_LOG=/logs/fittrackee.log
UPLOAD_FOLDER=/uploads
# gunicorn settings
# see https://docs.gunicorn.org/en/stable/settings.html#worker-processes
APP_WORKERS=2
GUNICORN_LOG=/logs/gunicorn.log
GUNICORN_TIMEOUT=300
GUNICORN_THREADS=4

# PostgreSQL
POSTGRES_USER=postgres
POSTGRES_PASSWORD=postgres
DATABASE_URL=postgresql://fittrackee:fittrackee@fittrackee-db:5432/fittrackee
#DATABASE_DISABLE_POOLING=

# Redis (required for API rate limits, email sending and user export)
#REDIS_URL=redis://redis:6379

# API rate limits (needs redis running)
#API_RATE_LIMITS="300 per 5 minutes"

# Emails
# since fittrackee 0.6.5, email sending can be disabled
# see documentation https://samr1.github.io/FitTrackee/installation.html#emails
#UI_URL=
#EMAIL_URL=
#SENDER_EMAIL=
#WORKERS_PROCESSES=2

# Workouts
#export TILE_SERVER_URL=
#export STATICMAP_SUBDOMAINS=
#export MAP_ATTRIBUTION=
#export DEFAULT_STATICMAP=False

# Weather
# available weather API providers: darksky, visualcrossing
#export WEATHER_API_PROVIDER=
#export WEATHER_API_KEY=
```

something is preventing the migration execution.

with `make migrate` command, the error should be displayed, for instance with an invalid path:

```shell
$ make migrate
Traceback (most recent call last):
  File "/usr/src/app/.venv/bin/ftcli", line 5, in <module>
    from fittrackee.cli import cli
  File "/usr/src/app/.venv/lib/python3.9/site-packages/fittrackee/__init__.py", line 34, in <module>
    logging.basicConfig(
  File "/usr/local/lib/python3.9/logging/__init__.py", line 2003, in basicConfig
    h = FileHandler(filename, mode,
  File "/usr/local/lib/python3.9/logging/__init__.py", line 1146, in __init__
    StreamHandler.__init__(self, self._open())
  File "/usr/local/lib/python3.9/logging/__init__.py", line 1175, in _open
    return open(self.baseFilename, self.mode, encoding=self.encoding,
FileNotFoundError: [Errno 2] No such file or directory: '<invalid_path>/logs/fittrackee.log'
make: *** [Makefile:13: migrate] Error 1
```

I never used docker-compose on a rasberry pi (my current FitTrackee instance is installed on a Raspberry Pi 4 but without docker (and with another OS)).

Did you delete the existing container before rebuilding it?

--- a/.env.example
+++ b/.env.example
@@ -5,13 +5,14 @@
 FITTRACKEE_LOG_DIR=
 
 # Application
+# directories are located in the docker container
 FLASK_APP=fittrackee
 FLASK_SKIP_DOTENV=1
 PORT=5000
 APP_SETTINGS=fittrackee.config.ProductionConfig
 APP_SECRET_KEY='please change me'
 APP_LOG=/logs/fittrackee.log
-UPLOAD_FOLDER=/uploads
+UPLOAD_FOLDER=/uploads  # must match docker-compose config
 # gunicorn settings
 # see https://docs.gunicorn.org/en/stable/settings.html#worker-processes
 APP_WORKERS=2

--- a/.env.example
+++ b/.env.example
@@ -18,6 +18,8 @@ APP_WORKERS=2
 GUNICORN_LOG=/logs/gunicorn.log
 GUNICORN_TIMEOUT=300
 GUNICORN_THREADS=4
+# workers settings
+WORKERS_PROCESSES=2
 
 # PostgreSQL
 POSTGRES_USER=postgres
@@ -37,7 +39,6 @@ DATABASE_URL=postgresql://fittrackee:fittrackee@fittrackee-db:5432/fittrackee
 #UI_URL=
 #EMAIL_URL=
 #SENDER_EMAIL=
-#WORKERS_PROCESSES=2
 
 # Workouts
 #export TILE_SERVER_URL=

--- a/README.md
+++ b/README.md
@@ -73,3 +73,15 @@ With default configuration (no `EMAIL_URL` set), email sending is disabled.
 Notes:
 - **Important**: all uncommented [variables](https://samr1.github.io/FitTrackee/installation.html#environment-variables) present in .env must be initialized. Otherwise, the application may not start.
 - If you just want to evaluate **FitTrackee**, ready to use docker files are available in **FitTrackee** repository (see [Documentation](https://samr1.github.io/FitTrackee/installation.html#docker)).
+
+Troubleshooting:
+- If installation or startup fails, check the environnement variables.  
+- The commands can be run separately when debugging, for instance:
+```shell
+$ make build
+$ make up
+$ make migrate
+$ make run
+```
+
+`make up` can be replaced with `docker-compose up` (without `detach` option to prevent containers from running in background and to display some errors).

--- a/README.md
+++ b/README.md
@@ -75,7 +75,7 @@ Notes:
 - If you just want to evaluate **FitTrackee**, ready to use docker files are available in **FitTrackee** repository (see [Documentation](https://samr1.github.io/FitTrackee/installation.html#docker)).
 
 Troubleshooting:
-- If installation or startup fails, check the environnement variables.  
+- If installation or startup fails, check the environment variables.  
 - The commands can be run separately when debugging, for instance:
 ```shell
 $ make build

